### PR TITLE
Remove some valid fields from the ignored ones

### LINKED ISSUE #318: allow_auto_merge, delete_branch_on_merge and other valid fields are marked as ignorableFields
allow_auto_merge, delete_branch_on_merge and other valid fields are marked as ignorableFields

https://github.com/github/safe-settings/blob/5dd8e7987f6bf685a44ad857d0c0cc3092cd7002/lib/plugins/repository.js#L31

### PR DESCRIPTION
**Pull request by glyph-se**
I recently discovered that there are still some valid fields in the `ignoredFields` list. We know need this to work in our organization. Similar to what I did in https://github.com/github/safe-settings/pull/344 I have now removed the others ones that I found.

This adds support for the following settings to the repository part of safe-settings: 'allow_merge_commit',   'allow_rebase_merge',   'delete_branch_on_merge',

Closes: #318

### CODE REVIEW

File: lib/plugins/repository.js [hunk 1]
```diff
@@ -26,9 +26,6 @@ const ignorableFields = [
   'open_issues',
   'watchers',
   'temp_clone_token',
-  'allow_merge_commit',
```
Comment (martinm82) [inline lib/plugins/repository.js:29]: What is the impact of this change?

Comment (glyph-se) [reply lib/plugins/repository.js:29]: It allows setting these three values from the settings.yml

**Comment by nicholasgibson2** [comment]
@martinm82 any updates on getting this merged? Would love to start leveraging these settings.

File: lib/plugins/repository.js (hunk 1 above)
Comment (martinm82) [reply lib/plugins/repository.js:29]: > @martinm82 any updates on getting this merged? Would love to start leveraging these settings.

Hi @nicholasgibson2, I am not the maintainer of this repository. @decyjphr could you please take a look at this PR.

**Comment by martinm82** [comment]
> @martinm82 any updates on getting this merged? Would love to start leveraging these settings.

Hi @nicholasgibson2, I am not the maintainer of this repository. @decyjphr could you please take a look at this PR.

**Pull request merged**